include/arch/x86/intel64/arch.h

```diff
@@ -93,11 +93,20 @@ struct x86_ssf {
 
 #endif /* _ASMLANGUAGE */
 
+#ifdef CONFIG_PCIE
+#define X86_RESERVE_IRQ(irq_p, name) \
+	static Z_DECL_ALIGN(uint8_t) name \
+	__in_section(_irq_alloc, static, name) __used = irq_p
```

Comment:
Seems like this should be using the Z_ITERABLE API, no?   It's just an array of static structs to be enumerated, right?

Reply:
I’d have wanted to use that, but it’s not structs here, rather plain uint8_t entries. I guess I could have wrapped them in structs but that felt a bit overkill.